### DISCUSSION TITLE
base on CS to match NOSQL DB |基于cs框架实现nosql数据库

### DISCUSSION THREAD

**Discussion by Formeeei** [community/community#130156] [Programming Help]
### Select Topic Area

Question

### Body

Hi！could you please tell me some idea about how to implement a NoSQL database based on the CS architecture?using java.Or can you tell me how can I find the relevant project?Thank you!
能不能提供一下怎样用java实现基于cs架构实现的nosql数据库的框架？或许哪里有开源指路一下[跪地][感激]

**Comment by HemalathaT2** [comment]
Using Cosmos DB

You can use the Azure Cosmos DB Java SDK to create a NoSQL database. Here's an example of how you can create a container and perform CRUD operations:
@Container(containerName = "products", autoCreateContainer = false)
public class Item {
    private String id;
    private String name;
    private Integer quantity;
    private Boolean sale;

    @PartitionKey
    private String category;
}

// Create an item in the container
Item item = new Item(
    "70b63682-b93a-4c77-aad2-65501347265f",
    "gear-surf-surfboards",
    "Yamba Surfboard",
    12,
    false
);
Item created_item = repository.save(item);

// Perform a point read operation
PartitionKey partitionKey = new PartitionKey("gear-surf-surfboards");
Optional<Item> existing_item = repository.findById("70b63682-b93a-4c77-aad2-65501347265f", partitionKey);
if (existing_item.isPresent()) {
    // Do something
}

// Perform a query over multiple items in a container
@Repository
public interface ItemRepository extends CosmosRepository<Item, String> {
    @Query("SELECT * FROM products p WHERE p.category = @category")
    List<Item> getItemsByCategory(@Param("category") String category);
}
